pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -0,0 +1,102 @@
+package machinehealthcheck
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const (
+	ControllerName string = "MachineHealthCheck"
+	managed        string = "aro.machinehealthcheck.managed"
+	enabled        string = "aro.machinehealthcheck.enabled"
+)
+
+type Reconciler struct {
+	arocli aroclient.Interface
+	dh     dynamichelper.Interface
+}
+
+func NewReconciler(arocli aroclient.Interface, dh dynamichelper.Interface) *Reconciler {
+	return &Reconciler{
+		arocli: arocli,
+		dh:     dh,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(enabled) {
+		return reconcile.Result{}, nil
+	}
+
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(managed) {
+		err := r.dh.EnsureDeleted(ctx, "MachineHealthCheck", "openshift-machine-api", "aro-machinehealthcheck")
+		if err != nil {
+			return reconcile.Result{RequeueAfter: time.Hour}, err
```

Comment:
We are using an hour wait also in `workaround_controller` and `checker_controller`, this should prevent this controller from being rate limited and not spam the deletion request.